handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -786,10 +788,60 @@ public SSLEngineResult unwrap(final ByteBuffer[] srcs, final ByteBuffer[] dsts)
         return unwrap(srcs, 0, srcs.length, dsts, 0, dsts.length);
     }
 
+    private ByteBuffer[] singleSrcBuffer(ByteBuffer src) {
+        singleSrcBuffer[0] = src;
+        return singleSrcBuffer;
+    }
+
+    private void resetSingleSrcBuffer() {
+        singleSrcBuffer[0] = null;
+    }
+
+    private ByteBuffer[] singleDstBuffer(ByteBuffer src) {
+        singleDstBuffer[0] = src;
+        return singleDstBuffer;
+    }
+
+    private void resetSingleDstBuffer() {
+        singleDstBuffer[0] = null;
+    }
+
     @Override
-    public SSLEngineResult unwrap(
+    public synchronized SSLEngineResult unwrap(
```

Comment:
@nmittler it does not matter as the super.unwrap(...) method just delegated to our already synchronized unwrap method.